### PR TITLE
Improve: SiFive peripheral device doumentation

### PR DESCRIPTION
**Pull request by T-K-233**
Add documentation on sifive peripheral devices and example code of how to add them to the ChipConfig.

**Related PRs / Issues**:

**Type of change**:
- [ ] Bug fix
- [ ] New feature
- [x] Other enhancement

**Impact**:
- [ ] RTL change
- [ ] Software change (RISC-V software)
- [ ] Build system change
- [x] Other

**Contributor Checklist**:
- [x] Did you set `main` as the base branch?
- [x] Is this PR's title suitable for inclusion in the changelog and have you added a `changelog:<topic>` label?
- [x] Did you state the type-of-change/impact?
- [x] Did you delete any extraneous prints/debugging code?
- [ ] Did you mark the PR with a `changelog:` label?
- [ ] (If applicable) Did you update the conda `.conda-lock.yml` file if you updated the conda requirements file?
- [ ] (If applicable) Did you add documentation for the feature?
- [ ] (If applicable) Did you add a test demonstrating the PR?

- [ ] (If applicable) Did you mark the PR as `Please Backport`?

### CODE REVIEW

File: docs/Generators/SiFive-Generators.rst [hunk 1]
```diff
@@ -11,29 +11,154 @@ Last-Level Cache Generator
...
     :language: scala
     :start-after: DOC include start: GPIORocketConfig
     :end-before: DOC include end: GPIORocketConfig
 
-Some of the devices in ``sifive-blocks`` (such as GPIO) may already have pre-defined config fragments within the Chipyard example project. You may be able to use these config fragments directly, but you should be aware of their addresses within the SoC address map.
+
+General Purpose I/Os (GPIO) Device
+----------------------------------
+
+GPIO device is a periphery device provided by ``sifive-blocks``. Each general-purpose I/O port has five 32-bit configuration registers (GPIOx_INPUT_EN, GPIOx_OUTPUT_EN, GPIOx_PUE GPIOx_DS, and GPIOx_XOR), two 32-bit data registers (GPIOx_INPUT_VAL and GPIOx_OUTPUT_VAL) and eight 32-bit interrupt control/status register (GPIOx_RISE_IE, GPIOx_RISE_IP, GPIOx_FALL_IE, GPIOx_FALL_IP, GPIOx_HIGH_IE, GPIOx_HIGH_IP, GPIOx_LOW_IE, GPIOx_LOW_IP). In addition, all GPIOs can have two 32-bit alternate function selection registers (GPIOx_IOF_EN and GPIOx_IOF_SEL).
```
Comment (jerryz123) [inline docs/Generators/SiFive-Generators.rst:37]: NIT. Can the GPIO variable names be in monospace font? : ``GPIOx_INPUT_EN``

File: docs/Generators/SiFive-Generators.rst [hunk 2]
```diff
@@ -11,29 +11,154 @@ Last-Level Cache Generator
...
+      new chipyard.config.WithGPIO(address = 0x10010000, width = 32) ++
+
+      // ...
+    )
+
+
+Universal Asynchronous Receiver/Transmitter (UART) Device
+----------------------------------------------------------
+
+UART device is a periphery device provided by ``sifive-blocks``. The UART offers a flexible means to perform Full-duplex data exchange with external devices. A very wide range of baud rates can be achieved through a fractional baud rate generator. The UART peripheral does not support hardware flow control or other modem control signals, or synchronous serial data transfers.
```
Comment (jerryz123) [inline docs/Generators/SiFive-Generators.rst:81]: Hardware flow control should be supported if the right parameter is enabled.

**Review by jerryz123 — CHANGES_REQUESTED** [review]
Minor changes, otherwise looks good!

File: docs/Generators/SiFive-Generators.rst (hunk 1 above)
Comment (T-K-233) [reply docs/Generators/SiFive-Generators.rst]: Decided to remove the specific register naming to match the other sections. Register map and naming should be found in each chip's spec doc.

**Review by jerryz123 — APPROVED** [review]

**Comment by jerryz123** [comment]
Thanks!

**Pull request merged**